lib/tests/diff.rs

```diff
@@ -0,0 +1,88 @@
+use img_hash::{HashType, ImageHash};
+use texture_synthesis as ts;
+
+macro_rules! diff_runs {
+    ($name:ident, $expected:expr, $gen:expr) => {
+        #[test]
+        fn $name() {
+            let expected_hash = ImageHash::from_base64($expected).expect("loaded hash");
+
+            let generated = $gen
+                // We always use a single thread to ensure we get consistent results
+                // across runs
+                .max_thread_count(1)
+                .build()
+                .unwrap()
+                .run(None);
+            let gen_img = generated.into_image();
+
+            let gen_hash = ImageHash::hash(&gen_img, 8, HashType::DoubleGradient);
+
+            if gen_hash != expected_hash {
+                let distance = expected_hash.dist_ratio(&gen_hash);
+                let txt_gen = gen_hash.to_base64();
+
+                assert_eq!($expected, txt_gen, "images hashes differed by {}", distance);
+            }
+        }
+    };
+}
+
+diff_runs!(single_example, "JKc2MqWo1iNWeJ856Ty6+a1M", {
```

Comment:
Should add some docs in the README and link here how to re-generate the hash and that one needs to update it when doing changes to the algorithm

Reply:
I put it into the diff.rs file itself as the README is more of a top level user guide, only people who actually make modifications would ever need to know about the tests.

Reply:
ok, yeah that should work well